src/utils/tensor.js

```diff
@@ -15,8 +15,10 @@ import {
 } from './maths.js';
 
 
-// @ts-ignore
-const DataTypeMap = new Map([
+/**
+ * @typedef {'bool'|'float32'|'float64'|'string'|'int8'|'uint8'|'int16'|'uint16'|'int32'|'uint32'|'int64'} DataType 
+ */
+const DataTypeMap = new Map( /** @type {[DataType, any][]} */([
```

Comment:
This is creating so much ASCII soup, why not just:

```js
const DataTypeMap = {
    bool: Uint8Array,
    float32: Float32Array,
    float64: Float64Array,
    string: Array, // string[]
    int8: Int8Array,
    uint8: Uint8Array,
    int16: Int16Array,
    uint16: Uint16Array,
    int32: Int32Array,
    uint32: Uint32Array,
    int64: BigInt64Array,
};
```

You get better typing for free without any casting hacks:

![image](https://github.com/xenova/transformers.js/assets/5236548/a307088d-f9aa-4190-bb15-4bbbb103248d)

Probably just went overboard with "Map is new holy grail" suggestions? We don't need any of that, the order doesn't matter, it won't contain keys like `toString`, we only need strings as key etc.

Reply:
I also suggest to keep "type variables" lower-case, it's the standard:

https://github.com/playcanvas/engine/blob/f9bac39b27f9df00d1ca6285a3c1c36b5af77fd4/src/framework/parsers/glb-parser.js#L151-L166